docs/src/components/ComponentDoc/ComponentControls/ComponentControlsCodeSandbox/ComponentControlsCodeSandbox.tsx

```diff
@@ -15,82 +24,86 @@ type ComponentControlsCodeSandboxProps = {
+
     return {
       exampleCode: props.exampleCode,
       examplePath: props.exampleLanguage === 'ts' ? '/example.tsx' : '/example.js',
-      sandboxUrl: props.exampleCode === state.exampleCode ? state.sandboxUrl : '',
+      sandboxUrl: shouldKeepState ? state.sandboxUrl : '',
+      state: shouldKeepState ? state.state : CodeSandboxState.Default,
     }
   }
 
   handleDeploy = (embedUrl: string, sandboxId: string) => {
     const { examplePath } = this.state
     const sandboxUrl = `https://codesandbox.io/s/${sandboxId}?module=${examplePath}`
 
-    this.setState({ sandboxUrl })
+    this.setState({ sandboxUrl, state: CodeSandboxState.Success })
   }
 
-  handleClick = e => {
-    const { sandboxUrl } = this.state
+  handleClick = (e: React.SyntheticEvent) => {
+    const { sandboxUrl, state } = this.state
+
     e.preventDefault()
-    window.open(sandboxUrl)
+
+    if (state === CodeSandboxState.Default) {
+      this.codeSandboxerRef.current.deployToCSB(e)
+      this.setState({ state: CodeSandboxState.Loading })
+
+      return
+    }
+
+    if (state === CodeSandboxState.Success) {
+      window.open(sandboxUrl)
+    }
   }
 
   render() {
-    const { exampleLanguage, exampleCode, exampleName } = this.props
-    const { examplePath, sandboxUrl } = this.state
+    const { children, exampleLanguage, exampleCode, exampleName } = this.props
+    const { examplePath, state } = this.state
 
     const main = exampleLanguage === 'ts' ? 'index.tsx' : 'index.js'
     const template = exampleLanguage === 'ts' ? 'create-react-app-typescript' : 'create-react-app'
 
-    if (sandboxUrl) {
-      return (
-        <Tooltip
-          trigger={<Icon name="checkmark" onClick={this.handleClick} />}
-          content="Click to open"
-        />
-      )
-    }
-
     return (
-      <CodeSandboxer
-        afterDeploy={this.handleDeploy}
-        examplePath={examplePath}
-        example={exampleCode}
-        name={exampleName}
-        providedFiles={{
-          [main]: { content: appTemplate },
-          'package.json': createPackageJson(main, exampleLanguage),
-        }}
-        skipRedirect
-        template={template}
-      >
-        {({ isLoading, isDeploying }) => {
-          const loading = isLoading || isDeploying
-          return (
-            <Tooltip
-              trigger={<Icon name={loading ? 'spinner' : 'connectdevelop'} />}
-              content={loading ? 'Exporting...' : 'CodeSandbox'}
-            />
-          )
-        }}
-      </CodeSandboxer>
+      <>
+        <CodeSandboxer
+          afterDeploy={this.handleDeploy}
+          examplePath={examplePath}
+          example={exampleCode}
+          name={exampleName}
+          providedFiles={{
+            [main]: { content: appTemplate },
+            'package.json': createPackageJson(main, exampleLanguage),
+          }}
+          skipRedirect
+          ref={this.codeSandboxerRef}
+          template={template}
+        >
+          {() => null}
```

Comment:
```suggestion
          // CodeSandboxer captures ref from children and binds an event handler directly to DOM, this hack allows to avoid this behavior
          {() => null}
```

Let's add a comment 💬